src/datasources/relay-api/gelato-api.service.spec.ts

```diff
@@ -0,0 +1,187 @@
+        chainId,
+        address,
+      });
+
+      expect(result).toEqual(0);
+    });
+  });
+
+  describe('relay', () => {
+    it('should relay the payload', async () => {
+      const chainId = faker.string.numeric();
+      const address = faker.finance.ethereumAddress() as Hex;
+      const data = faker.string.hexadecimal() as Hex;
+      const apiKey = faker.string.sample();
+      const taskId = faker.string.alphanumeric();
+
+      fakeConfigurationService.set(`gelato.apiKey.${chainId}`, apiKey);
+
+      mockGelatoClient.sponsoredCall.mockResolvedValue({ taskId });
+
+      const result = await target.relay({
+        chainId,
+        to: address,
+        data,
+      });
+
+      expect(result).toEqual({ taskId });
+      expect(mockGelatoClient.sponsoredCall).toHaveBeenCalledWith(
+        {
+          chainId: BigInt(chainId),
+          data,
+          target: address,
+        },
+        apiKey,
+        {
+          gasLimit: undefined,
+        },
+      );
+    });
+
+    it('should add a gas buffer if a gas limit is provided', async () => {
+      const chainId = faker.string.numeric();
+      const address = faker.finance.ethereumAddress() as Hex;
+      const data = faker.string.hexadecimal() as Hex;
+      const gasLimit = faker.number.bigInt();
+      const apiKey = faker.string.sample();
+
+      fakeConfigurationService.set(`gelato.apiKey.${chainId}`, apiKey);
+
+      await target.relay({
+        chainId,
+        to: address,
+        data,
+        gasLimit,
+      });
+
+      expect(mockGelatoClient.sponsoredCall).toHaveBeenCalledWith(
+        {
+          chainId: BigInt(chainId),
+          data,
+          target: address,
+        },
+        apiKey,
+        {
+          gasLimit: gasLimit + BigInt(150_000),
+        },
+      );
+    });
+
+    it('should throw if there is no API key preset', async () => {
+      const chainId = faker.string.numeric();
+      const address = faker.finance.ethereumAddress() as Hex;
+      const data = faker.string.hexadecimal() as Hex;
+
+      await expect(
+        target.relay({
+          chainId,
+          to: address,
+          data,
+        }),
+      ).rejects.toThrow();
+    });
+
+    it('should increment the count after relaying', async () => {
+      const setSpy = jest.spyOn(fakeCacheService, 'set');
+
+      const chainId = faker.string.numeric();
+      const address = faker.finance.ethereumAddress() as Hex;
+      const data = faker.string.hexadecimal() as Hex;
+      const apiKey = faker.string.sample();
+
+      fakeConfigurationService.set(`gelato.apiKey.${chainId}`, apiKey);
+
+      await target.relay({
+        chainId,
+        to: address,
+        data,
+      });
+
+      expect(setSpy).toHaveBeenCalledWith(
+        new CacheDir(`${chainId}_relay_${address}`, ''),
+        '1',
+      );
+    });
+
+    it('should not fail the relay if incrementing the count fails', async () => {
+      const setSpy = jest.spyOn(fakeCacheService, 'set');
```

Comment:
Same here. If the goal is to measure the number of calls probably a fake implementation is the wrong tool for this test.

Reply:
This is needed to throw setting the cache. I've moved it to where it's used in 641026000723aeba2b9c4abce2179b1b7d54ecd1 and added a comment. Let me know your thoughts.

Reply:
While I understand the use-case I think it can be confusing to mock/change the behaviour of the fake object via Jest.

We should either extend the fake object to provide what we need for the tests or use a mocked instance. This works and is mostly a minor comment so no need to address it now.

Reply:
I overlooked this comment before merging. I will pass a mocked instance instead in a separate PR.